### PR TITLE
Update iced to 0.14.0-dev

### PR DESCRIPTION
**Pull request by englut**
Found an issue with touchpad scrolling being reversed, checked out the CHANGELOG for iced, and noticed 0.13.1 fixes the issue that was introduced in 0.13.0:

https://github.com/iced-rs/iced/pull/2583

Updated iced to `0.14.0-dev`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ ipc = { version = "0.1.0", path = "ipc" }
 notify-rust = "4"
 chrono = { version = "0.4", features = ['serde'] }
 fern = "0.6.1"
-iced = { version = "0.13.0-dev", features = [
+iced = { version = "0.13.1", features = [
```
Comment (casperstorm) [inline Cargo.toml:22]: Add `0.14.0-dev` here instead, as we always target latest master commit.

File: Cargo.toml [hunk 2]
```diff
@@ -61,5 +61,5 @@ windows_exe_info = "0.4"
 members = ["data", "ipc", "irc", "irc/proto"]
 
 [patch.crates-io]
-iced = { git = "https://github.com/iced-rs/iced", rev = "8fb939b5a920e0cd836dbdd24c948f8f2512fc7e" }
-iced_core = { git = "https://github.com/iced-rs/iced", rev = "8fb939b5a920e0cd836dbdd24c948f8f2512fc7e" }
+iced = { git = "https://github.com/iced-rs/iced", rev = "9bfbd7cda79aceef2d115b8bb35e8f3257dcabf2" }
+iced_core = { git = "https://github.com/iced-rs/iced", rev = "9bfbd7cda79aceef2d115b8bb35e8f3257dcabf2" }
```
Comment (casperstorm) [inline Cargo.toml:65]: Lets grab latest master commit.

**Comment by neilalexander** [comment]
FYI on macOS:
```
error[E0432]: unresolved imports `iced::advanced::graphics::futures::subscription::MacOS`, `iced::advanced::graphics::futures::subscription::PlatformSpecific`
 --> src/url.rs:8:66
  |
8 |     use iced::advanced::graphics::futures::subscription::{Event, MacOS, PlatformSpecific};
  |                                                                  ^^^^^  ^^^^^^^^^^^^^^^^ no `PlatformSpecific` in `subscription`
  |                                                                  |
  |                                                                  no `MacOS` in `subscription`
  |
  = help: consider importing one of these variants instead:
          crate::url::iced::keyboard::key::NativeCode::MacOS
          iced::keyboard::key::NativeCode::MacOS
  = help: consider importing one of these structs instead:
          crate::url::iced::window::settings::PlatformSpecific
          iced::window::settings::PlatformSpecific

error[E0599]: no variant or associated item named `PlatformSpecific` found for enum `iced::advanced::subscription::Event` in the current scope
  --> src/url.rs:35:32
   |
35 |                         Event::PlatformSpecific(event) => match event {
   |                                ^^^^^^^^^^^^^^^^ variant or associated item not found in `Event`

Some errors have detailed explanations: E0432, E0599.
For more information about an error, try `rustc --explain E0432`.
```

**Comment by casperstorm** [comment]
@hecrj is cooking on iced, so it moves fast! 🥳 
Once we are up-to-date with master we can merge this again.

File: Cargo.toml (hunk 1 above)
Comment (englut) [reply Cargo.toml:22]: @casperstorm I'm getting these errors when building against `0.14.0-dev` and latest master commit, looks like the color struct has moved, I'll do some tweaking!

```
note: `iced_core::color::Color` is defined in crate `iced_core`
   --> /home/quaff/.cargo/registry/src/index.crates.io-6f17d22bba15001f/iced_core-0.13.2/src/color.rs:5:1
    |
5   | pub struct Color {
    | ^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `iced_core` are being used?
```

**Comment by englut** [comment]
@casperstorm I've moved iced back to `0.13.0-dev` but using commit `3a9139e64f37f3223be1a417357123c472ce7c27`. This is the most up-to-date master commit that doesn't introduce breaking changes to our code base. 

When trying to bump the commit versions further: `MacOS` and `PlatformSpecific` isn't exported (or even showing up in the compiled `iced_futures` library) via `iced::advanced::graphics::futures::subscription`. The `Event` struct that's exported here doesn't have `MacOS` specific events either. Which makes me think there's some kind of platform build that's excluding these things, I could be wrong though. Maybe you or @tarkah could take a look at this one.

If I switch to `0.14.0-dev` there's all sorts of errors that I haven't quite figured out what they mean yet. There's a bunch of mismatched types relating to `iced_core::size::Size`:

```
error[E0308]: mismatched types
   --> src/main.rs:230:28
    |
230 |             min_size: Some(window::MIN_SIZE),
    |                       ---- ^^^^^^^^^^^^^^^^ expected `iced::Size`, found `iced_core::size::Size`
    |                       |
    |                       arguments to this enum variant are incorrect
    |
    = note: `iced_core::size::Size` and `iced::Size` have similar names, but are actually distinct types
note: `iced_core::size::Size` is defined in crate `iced_core`
   --> /Users/ming/.cargo/registry/src/index.crates.io-6f17d22bba15001f/iced_core-0.13.2/src/size.rs:5:1
    |
5   | pub struct Size<T = f32> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^
note: `iced::Size` is defined in crate `iced_core`
   --> /Users/ming/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/b78b8bc/core/src/size.rs:5:1
    |
5   | pub struct Size<T = f32> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `iced_core` are being used?
help: the type constructed contains `iced_core::size::Size` due to the type of the argument passed
   --> src/main.rs:230:23
    |
230 |             min_size: Some(window::MIN_SIZE),
    |                       ^^^^^----------------^
    |                            |
    |                            this argument influences the type of `Some`
note: tuple variant defined here
   --> /Users/ming/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/option.rs:582:5
    |
582 |     Some(#[stable(feature = "rust1", since = "1.0.0")] T),
    |     ^^^^
```

And there's multiple issues relating to `iced_core::color::Color`:

```
error[E0277]: the trait bound `iced::Color: From<iced_core::color::Color>` is not satisfied
   --> src/widget/message_content.rs:91:37
    |
90  |   ...                   .color_maybe(
    |                          ----------- required by a bound introduced by this call
91  | / ...                       formatting
92  | | ...                           .fg
93  | | ...                           .and_then(|color| color.into_iced(theme.colors())),
    | |________________________________________________________________________________^ the trait `From<iced_core::color::Color>` is not implemented for `iced::Color`, which is required by `iced_core::color::Color: Into<iced::Color>`
    |
    = help: the following other types implement trait `From<T>`:
              `iced::Color` implements `From<Alpha<palette::rgb::Rgb, f32>>`
              `iced::Color` implements `From<[f32; 3]>`
              `iced::Color` implements `From<[f32; 4]>`
              `iced::Color` implements `From<palette::rgb::Rgb>`
    = note: required for `iced_core::color::Color` to implement `Into<iced::Color>`
note: required by a bound in `iced::advanced::text::Span::<'a, Link, Font>::color_maybe`
   --> /Users/ming/.cargo/git/checkouts/iced-f01cba4d5e61fd0a/b78b8bc/core/src/text.rs:330:53
    |
330 |     pub fn color_maybe(mut self, color: Option<impl Into<Color>>) -> Self {
    |                                                     ^^^^^^^^^^^ required by this bound in `Span::<'a, Link, Font>::color_maybe`
```

I'm willing to spend more time on this, but I fear I maybe be stabbing in the wrong places. Maybe you or @tarkah may spot some quick fixes for these 🙏

**Comment by hecrj** [comment]
> If I switch to 0.14.0-dev there's all sorts of errors that I haven't quite figured out what they mean yet. There's a bunch of mismatched types relating to `iced_core::size::Size`

Running `cargo update` should fix it.

**Comment by englut** [comment]
> > If I switch to 0.14.0-dev there's all sorts of errors that I haven't quite figured out what they mean yet. There's a bunch of mismatched types relating to `iced_core::size::Size`
> 
> Running `cargo update` should fix it.

Still getting those errors even after running `cargo clean` and `cargo update` 🙏

**Comment by hecrj** [comment]
You didn't update `iced_core` in `data/Cargo.toml`.

**Comment by neilalexander** [comment]
Build worked fine on my Apple Silicon Mac now BTW.

**Review by casperstorm — APPROVED** [review]

**Pull request merged**